android-interop-testing/app/src/androidTest/java/io/grpc/android/integrationtest/TesterActivityTest.java

```diff
@@ -0,0 +1,58 @@
+package io.grpc.android.integrationtest;
+
+import android.os.Bundle;
+import android.test.ActivityInstrumentationTestCase2;
+import android.test.InstrumentationTestRunner;
+import junit.framework.Assert;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The Activity test of our App, used to trigger the integration test from commandline.
+ */
+public class TesterActivityTest extends ActivityInstrumentationTestCase2<TesterActivity> {
+
+  private String host;
+  private int port = 8030;
+  private String result;
+  private String serverHostOverride;
+  private String testCase;
+  private boolean useTls;
+  private boolean useTestCa;
+
+  public TesterActivityTest() {
+    super(TesterActivity.class);
+  }
+
+  public void setUp() throws Exception {
+    super.setUp();
+    getActivity();
+    Bundle args = ((InstrumentationTestRunner) getInstrumentation()).getArguments();
+    host = args.getString("server_host", "");
+    port = Integer.parseInt(args.getString("server_port", "0"));
+    assertTrue("Please specify server_host and server_port.", !host.isEmpty() && port != 0);
+    serverHostOverride = args.getString("server_host_override", null);
+    testCase = args.getString("test_case", "empty_unary");
+    useTls = Boolean.parseBoolean(args.getString("use_tls", "true"));
+    useTestCa = Boolean.parseBoolean(args.getString("use_test_ca", "false"));
+  }
+
+  public void testGrpc() throws Exception {
+    final CountDownLatch finished = new CountDownLatch(1);
+
+    new GrpcTestTask(testCase, host, port, serverHostOverride, useTls, useTestCa,
```

Comment:
For the interop testing, are you suggesting doing the test in like the onCreate() method? that seems a little weird, and from the perspective of our interop test framework, checking "whether an Android instrumentation test passed" is more straightforward than checking "whether an Android activity is started".

For the Unit test, I tried but it turned out that it's not trivial to run grpc server on Android, here are the problems I'm facing:
1. We need to depends on grpc:grpc-interop-testing which provides a dependency to google-auth-library-* and in the end depends on javaee-api which is not allowed to use on Android. I expect this can be resolved by proguard, but there were other issues introduced after enabling proguard and I'm still fighting with them.
2. The netty server/SslContextBuilder API can only accept File object for certificate chain, while Android resources can only be accessed through InputStream, so at the beginning of the test, we have to store the PEM into some File, then pass to server API, which is not pretty.
3. Since I haven't got the server run, I'm not sure whether we can start the grpc server directly on Android, there may be more issues need to handle.

So I prefer postponing the Unit test from this PR.

And for the Provider installing, I'm following the example given by the android developer training:
https://developer.android.com/training/articles/security-gms-provider.html